### PR TITLE
Python 3: port tests in clear-site-data etc.

### PR DESCRIPTION
**Pull request by ziransun**
It includes porting tests for the following directories:
  clear-site-data
  signed-exchange
  web-animation
  scroll-to-text-fragment
  preload

### CODE REVIEW

File: clear-site-data/support/controlled-endpoint.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 def main(request, response):
-    return ([("Content-Type", "text/html")],
-            "FROM_NETWORK")
\ No newline at end of file
+    return ([(b"Content-Type", b"text/html")],
+            u"FROM_NETWORK")
```
Comment (annevk) [inline clear-site-data/support/controlled-endpoint.py:3]: Could you add a newline here if it's not too much trouble?

File: scroll-to-text-fragment/stash.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import time
 
 def main(request, response):
-    key = request.GET.first("key")
+    key = request.GET.first(b"key")
 
-    if request.method == "POST":
+    if request.method == u"POST":
```
Comment (annevk) [inline scroll-to-text-fragment/stash.py:6]: This is a bit weird as I noted in the other PR, but I guess it's an upstream problem?

**Review by annevk — APPROVED** [review]

File: scroll-to-text-fragment/stash.py (hunk 2 above)
Comment (ziransun) [reply scroll-to-text-fragment/stash.py:6]: Added. Thank you!

Comment (stephenmcgruer) [reply scroll-to-text-fragment/stash.py:6]: I think you meant to leave the "Added. Thank you!" reply on the other comment Ziran?

For this, cc @Hexcles as an FYI

File: clear-site-data/support/controlled-endpoint.py (hunk 1 above)
Comment (ziransun) [reply clear-site-data/support/controlled-endpoint.py]: Added.

File: scroll-to-text-fragment/stash.py (hunk 2 above)
Comment (ziransun) [reply scroll-to-text-fragment/stash.py:6]: @stephenmcgruer: yes, I meant to leave the reply to the other comment. Thanks for spotting this :).

**Comment by stephenmcgruer** [comment]
Fails Chrome stability check for `preload/modulepreload.html`, but I am able to reproduce without this patch so admin-merging.

File: scroll-to-text-fragment/stash.py (hunk 2 above)
Comment (Hexcles) [reply scroll-to-text-fragment/stash.py:6]: https://github.com/web-platform-tests/wpt/issues/23362 is the tracking issue.

**Pull request merged**